crates/astria-sequencer/src/mempool/mod.rs

```diff
@@ -417,16 +417,16 @@ impl Mempool {
 
             if demotion_txs.is_empty() {
                 // nothing to demote, check for transactions to promote
-                let highest_pending_nonce = pending
+                let pending_nonce = pending
                     .pending_nonce(address)
-                    .map_or(current_nonce, |nonce| nonce.saturating_add(1));
+                    .map_or(current_nonce, |nonce| nonce);
```

Comment:
This was a misnomer previously,  and the addition done papering over that our pending nonce math was wrong.

Moved the addition into the mempool so the math is correct throughout codebase.